### PR TITLE
Fixing eager loading issue in admin/moderation/reports

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [x] Bug Fix

## Description

If we open http://localhost:3000/admin/moderation/reports in locally, it fails to load the page with following error:

```
00:58:04 web.1       | GET /admin/moderation/reports
00:58:04 web.1       | AVOID eager loading detected
00:58:04 web.1       |   FeedbackMessage => [:reporter]
00:58:04 web.1       |   Remove from your query: .includes([:reporter])
00:58:04 web.1       | Call stack
00:58:04 web.1       |   /forem/app/views/admin/feedback_messages/_feedback_message.html.erb:146:in `_app_views_admin_feedback_messages__feedback_message_html_erb___2929576808175770137_571260'
00:58:04 web.1       |  /forem/app/views/admin/feedback_messages/index.html.erb:34:in `block (2 levels) in _app_views_admin_feedback_messages_index_html_erb__3034821833808733872_570820'
00:58:04 web.1       |   /forem/app/views/admin/feedback_messages/index.html.erb:33:in `block in _app_views_admin_feedback_messages_index_html_erb__3034821833808733872_570820'
00:58:04 web.1       |   /forem/app/views/admin/feedback_messages/index.html.erb:32:in `_app_views_admin_feedback_messages_index_html_erb__3034821833808733872_570820'

```

This PR fixes this error by removing `reporter` from eager loading.

## Related Tickets & Documents

- None

## QA Instructions, Screenshots, Recordings

Error + Solution Video: https://www.loom.com/share/600d4b029f8c49babe98d805a82bf78e

### UI accessibility concerns?

No

## Added/updated tests?

- [x] No

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated

### CODE REVIEW

**Comment by jaw6** [comment]
Any suggestions on how to QA this? I don't see bullet errors on `main`, so I'm thinking there might be a particular data scenario I would need to trigger it?

**Comment by lightalloy** [comment]
I couldn't reproduce this error too, but checked the code: it's interesting that we do have refer to the `feedback_message.reporter` in this view: https://github.com/forem/forem/blob/e4e1f4d96d5abc8a4ab15069d4b2709b10de9522/app/views/admin/feedback_messages/_feedback_message.html.erb#L130

So, if we remove `:reporter` from `includes` and create a feedback message that has the reporter, then go to the described page, we'll get an error from bullet:

```
USE eager loading detected
  FeedbackMessage => [:reporter]
  Add to your query: .includes([:reporter])
Call stack
  /home/light/sites/dev.to/app/views/admin/feedback_messages/_feedback_message.html.erb:130:in `_app_views_admin_feedback_messages__feedback_message_html_erb___3216089565245649644_508000'
```

So, we can't just remove the `includes`. I'll look into this later and try to find why bullet could complain on this `include`

**Comment by lightalloy** [comment]
I could reproduce (on another computer) on a main branch with this data:

```
=> [#<FeedbackMessage:0x000055c202980750
  id: 1,
  affected_id: nil,
  category: "spam",
  created_at: Thu, 25 Mar 2021 09:56:47.155902000 UTC +00:00,
  feedback_type: "spam",
  message: "Itaque tempore molestiae et.",
  offender_id: nil,
  reported_url: nil,
  reporter_id: 11,
  status: "Open",
  updated_at: Thu, 25 Mar 2021 09:56:47.155902000 UTC +00:00>,
 #<FeedbackMessage:0x00007eff7df6c3f8
  id: 2,
  affected_id: nil,
  category: "harassment",
  created_at: Thu, 25 Mar 2021 09:56:47.161301000 UTC +00:00,
  feedback_type: "abuse-reports",
  message: "Fugit qui dolores inventore.",
  offender_id: nil,
  reported_url: "example.com",
  reporter_id: 1,
  status: "Open",
  updated_at: Thu, 25 Mar 2021 09:56:47.161301000 UTC +00:00>]
```

Interestingly, these records have `reporter_id`, so we do use the loaded association.
Maybe it's related to the [bullet issue](https://github.com/flyerhzm/bullet/issues/641)

**Comment by rt4914** [comment]
@jaw6 @lightalloy 
Can you help in figuring out what can be the next steps here for me?

**Comment by jaw6** [comment]
@rt4914 I would recommend adding `reporter` to the bullet configuration [here](https://github.com/forem/forem/blob/a6030022f6bae2d75d021a53bb2684ec25989380/config/environments/development.rb#L135-L145) as (another) `unused_eager_loading`. It should be fairly straight-forward, but let me know if you want to pair on it, sometimes nice to have an extra set of eyes!

File: app/controllers/admin/feedback_messages_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class FeedbackMessagesController < Admin::ApplicationController
     layout "admin"
 
     def index
-      @q = FeedbackMessage.includes(:reporter, :offender, :affected)
+      @q = FeedbackMessage.includes(:offender, :affected)
```
Comment (jaw6) [inline app/controllers/admin/feedback_messages_controller.rb:6]: I mentioned this in a comment previously, but I think the system wants me to formally request changes, so... I think instead we should add `reporter` to the bullet configuration [here](https://github.com/forem/forem/blob/a6030022f6bae2d75d021a53bb2684ec25989380/config/environments/development.rb#L135-L145) as (another) `unused_eager_loading`. 

It should be fairly straight-forward, but let me know if you want to pair on it, sometimes nice to have an extra set of eyes!

**Review by jaw6 — CHANGES_REQUESTED** [review]

**Comment by rt4914** [comment]
Thanks @jaw6 Will apply these changes on Monday.

File: app/controllers/admin/feedback_messages_controller.rb (hunk 1 above)
Comment (rt4914) [reply app/controllers/admin/feedback_messages_controller.rb]: I have applied this change. Thanks @jaw6 for clarification.

`Bullet.add_safelist(type: :unused_eager_loading, class_name: "FeedbackMessage", association: :reporter)`

**Review by Zainahad19 — APPROVED** [review]

**Review by Zainahad19 — APPROVED** [review]

**Comment by Zainahad19** [comment]
Duplicate of #

> 
<details><summary>Details</summary>
<p>



</p>
</details>

**Review by klardotsh — COMMENTED** [review]
I don't have enough context to provide any meaningful review here, so I'm tagging myself out from review on this (or hoping that this comment-review fulfills GitHub's request for my comment).

**Review by lightalloy — APPROVED** [review]

**Pull request merged**